### ISSUE TITLE
CA2213: False positive with disposing pattern

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#6202]
### Analyzer

**Diagnostic ID**: [CA2213](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2213): `Disposable fields should be disposed`

### Describe the bug

When implementing the dispose pattern, calls to sub-sequent `Dispose` are wrapped inside a `if (disposing)` branch which would cause the field to be considered as not disposed in all paths. Because this is a recommended pattern, the analyzer should be tweaked to consider this case as always disposed case.

See https://github.com/dotnet/roslyn-analyzers/pull/6173#issuecomment-1273068397

### Steps To Reproduce

```csharp
public class BaseClass : IDisposable
{
    private bool disposedValue;
	private IDisposable somethingDisposable;

    protected virtual void Dispose(bool disposing)
    {
        if (!disposedValue)
        {
            if (disposing)
            {
                // TODO: dispose managed state (managed objects)
                somethingDisposable.Dispose();
            }
            // TODO: free unmanaged resources (unmanaged objects) and override finalizer
            // TODO: set large fields to null
            disposedValue = true;
        }
    }

    public void Dispose()
    {
        // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
        Dispose(disposing: true);
        GC.SuppressFinalize(this);
    }
}
```

**Comment by Evangelink** [comment]
@dominoFire @mavasani I will try to get to this fix as-soon-as possible.
